### PR TITLE
Add support for .test ENS domains

### PR DESCRIPTION
**Pull request by bitpshr**
This PR adds support for `.test` domains when using the Ropsten network and is based on #4405. By side effect, this PR also fixes an issue where the resolver was never updated after initialization even if the network changed.

http://metamask-ropsten.test/

**Note:** This should not be merged until #4405 has landed.

### CODE REVIEW

File: app/scripts/lib/resolver.js [hunk 1]
```diff
@@ -61,7 +61,7 @@ function getRegistrar (type) {
 module.exports.resolve = function (name, provider) {
   const path = name.split('.')
   const tld = path[path.length - 1]
-  if (tld === 'eth') {
+  if (tld === 'eth' || tld === 'test') {
```
Comment (jennypollack) [inline app/scripts/lib/resolver.js]: what is tld?

Comment (bitpshr) [reply app/scripts/lib/resolver.js]: TLD === "top-level domain" ([reference](https://en.wikipedia.org/wiki/Top-level_domain)). I updated the variable name so it's not as cryptic.

**Review by danfinlay — CHANGES_REQUESTED** [review]
Simple QA pass gave me this on start up:

```
TypeError: Cannot read property 'onBeforeRequest' of null
    at module.exports (ipfsContent.js:39)
```

**Review by danfinlay — APPROVED** [review]
Nuked deps, works for me! Looks good.

**Review by danfinlay — DISMISSED** [review]

**Review by danfinlay — APPROVED** [review]

**Pull request merged**